DependencyInjection/MonologExtension.php

```diff
@@ -399,32 +400,60 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
             $nestedHandlerId = $this->getHandlerId($handler['handler']);
             $this->markNestedHandler($nestedHandlerId);
 
+            $isLegacyBridge = !\class_exists(SwitchUserTokenProcessor::class);
+
             if (isset($handler['activation_strategy'])) {
                 $activation = new Reference($handler['activation_strategy']);
             } elseif (!empty($handler['excluded_404s'])) {
                 if (class_exists(HttpCodeActivationStrategy::class)) {
                     @trigger_error('The "excluded_404s" option is deprecated in MonologBundle since version 3.4.0, you should rely on the "excluded_http_codes" option instead.', E_USER_DEPRECATED);
                 }
-                $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
-                    new Reference('request_stack'),
-                    $handler['excluded_404s'],
-                    $handler['action_level']
-                ]);
+                if ($isLegacyBridge) {
+                    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
+                        new Reference('request_stack'),
+                        $handler['excluded_404s'],
+                        $handler['action_level']
+                    ]);
+                } else {
+                    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
+                        new Reference('request_stack'),
+                        $handler['excluded_404s'],
+                        new Definition('Monolog\Handler\FingersCrossed\ErrorLevelActivationStrategy', [
+                            $handler['action_level']
+                        ])
+                    ]);
+                }
```

Comment:
IMHO all this can be simplified to
```diff
+ $activation = $handler['action_level'];
+ if (class_exists(SwitchUserTokenProcessor::class)) {
+    $action = new Definition(ErrorLevelActivationStrategy::class, [
+       $activation
+   ])
+ }

  // ...

  if (...) {
    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
        new Reference('request_stack'),
        $handler['excluded_404s'],
-       $handler['action_level']
+       $activation
    ]);
    $activation = new Reference($handlerId.'.not_found_strategy');
  } elseif (...) {
    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\HttpCodeActivationStrategy', [
        new Reference('request_stack'),
        $handler['excluded_http_codes'],
-       $handler['action_level']
+       $activation
    ]);
    $activation = new Reference($handlerId.'.http_code_strategy');
-  } else {
-    $activation = $handler['action_level'];
  }
```